script/upgrades/eXOF/eXOFChecks.verify.sol

```diff
@@ -0,0 +1,517 @@
+      .getWhitelist();
+
+    if (!Arrays.contains(feeCurrencyWhitelist, eXOF)) {
+      revert("eXOF has not been added to the fee currency whitelist.");
+    }
+
+    console.log("🟢 eXOF has been added to the fee currency whitelist");
+  }
+
+  function verifyExchanges() internal {
+    eXOFConfig.eXOF memory config = eXOFConfig.get(contracts);
+
+    console.log("\n== Verifying exchanges ==");
+
+    verifyPoolExchange(config);
+    verifyPoolConfig(config);
+    verifyTradingLimits(config);
+  }
+
+  function verifyPoolExchange(eXOFConfig.eXOF memory config) internal view {
+    bytes32[] memory exchanges = BiPoolManager(biPoolManagerProxy).getExchangeIds();
+
+    // check configured pools against the config
+    require(
+      exchanges.length == config.pools.length + 7,
+      "Number of expected pools does not match the number of deployed pools."
+    );
+
+    for (uint256 i = 0; i < config.pools.length; i++) {
+      bytes32 exchangeId = getExchangeId(config.pools[i].asset0, config.pools[i].asset1, config.pools[i].isConstantSum);
+
+      IBiPoolManager.PoolExchange memory pool = BiPoolManager(biPoolManagerProxy).getPoolExchange(exchangeId);
+      Config.Pool memory poolConfig = config.pools[i];
+
+      // verify asset0 of the deployed pool against the config
+      if (pool.asset0 != poolConfig.asset0) {
+        console.log(
+          "The asset0 of deployed pool: %s does not match the expected asset0: %s.",
+          pool.asset0,
+          poolConfig.asset0
+        );
+        revert("asset0 of pool does not match the expected asset0. See logs.");
+      }
+
+      // verify asset1 of the deployed pool against the config
+      if (pool.asset1 != poolConfig.asset1) {
+        console.log(
+          "The asset1 of deployed pool: %s does not match the expected asset1: %s.",
+          pool.asset1,
+          poolConfig.asset1
+        );
+        revert("asset1 of pool does not match the expected asset1. See logs.");
+      }
+
+      if (poolConfig.isConstantSum) {
+        if (address(pool.pricingModule) != constantSum) {
+          console.log(
+            "The pricing module of deployed pool: %s does not match the expected pricing module: %s.",
+            address(pool.pricingModule),
+            constantSum
+          );
+          revert("pricing module of pool does not match the expected pricing module. See logs.");
+        }
+      } else {
+        if (address(pool.pricingModule) != constantProduct) {
+          console.log(
+            "The pricing module of deployed pool: %s does not match the expected pricing module: %s.",
+            address(pool.pricingModule),
+            constantProduct
+          );
+          revert("pricing module of pool does not match the expected pricing module. See logs.");
+        }
+      }
+      // verify asset0 is always a stable asset
+      require(
+        pool.asset0 == cUSD || pool.asset0 == cEUR || pool.asset0 == cBRL || pool.asset0 == eXOF,
```

Comment:
now that we have the `Arrays.contains` function we could also put all of these in an array and use that instead. Similar for the collateral assets.

Reply:
🙅 We could, and the function would make the code more concise. However, switching to it doesn't add much and would make the intention of the require statement less clear, as we would need to introduce an additional data structure to hold the stables. The current implementation with multiple OR conditions is relatively clear and directly shows the intention of the code.